tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -518,3 +524,407 @@ def test_io_data_server_pod_restart(self):
+                                           restore_params={
+                                               "deployment_name": self.pod_dict.get(pod_prefix)[-2],
+                                               "deployment_backup": self.deployment_backup,
+                                               "num_replica": self.pod_dict.get(pod_prefix)[2],
+                                               "set_name": self.pod_dict.get(pod_prefix)[1]},
+                                           clstr_status=True)
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+            LOGGER.info("Successfully restored pod by %s way", self.restore_method)
+        LOGGER.info("Step 9: Successfully started data and server pod and cluster is online.")
+        self.restore_pod = False
+        event.clear()
+        thread_rd.join()
+
+        LOGGER.info("Step 10: Verify status for In-flight READs/Verify DI")
+        responses_rd = dict()
+        while len(responses_rd) != 2:
+            responses_rd = output_rd.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in responses_rd["pass_res"])
+        fail_logs = list(x[1] for x in responses_rd["fail_res"])
+        all_logs = pass_logs + fail_logs
+        resp = self.ha_obj.check_s3bench_log(file_paths=all_logs)
+        assert_utils.assert_false(len(resp[1]), "READs/VerifyDI after pod restart are "
+                                  "expected to pass. Logs which contain failures:"
+                                  f" {resp[1]}. Logs for IOs when event was clear: {pass_logs}. "
+                                  f"Logs for IOs when event was set: {fail_logs}")
+        LOGGER.info("Step 10: Verified status for In-flight READs/Verify DI")
+
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 11: Create new IAM user and buckets, Perform WRITEs-READs-Verify "
+                        "with variable object sizes after pod restart")
+            users = self.mgnt_ops.create_account_users(nusers=1)
+            self.test_prefix = f'test-45536-restart-{int(perf_counter_ns())}'
+            self.s3_clean.update(users)
+        else:
+            LOGGER.info("Step 11: Create new objects, Perform WRITEs-READs-Verify with "
+                        "variable object sizes after pod restart")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 11: Performed WRITEs-READs-Verify with variable sizes objects after pod "
+                    "restart")
```

Comment:
Are we reading-verifying data written in healthy mode?

Reply:
we are doing read/verify in second last step in background, in last step we are doing IOs